app/controllers/dashboard/repositories.js

```diff
@@ -99,6 +102,20 @@ export default Ember.Controller.extend({
     selectOrg(org) {
       let login = Ember.isBlank(org) ? undefined : org.get('login');
       return this.set('account', login);
+    },
+    star(repo) {
+      this.get('ajax').ajax(`/v3/repo/${repo.get('id')}/star`, 'POST')
+        .then(() => {
+          this.get('flashes')
+            .success(`You successfully starred ${repo.get('slug')}`);
+        });
+    },
+    unstar(repo) {
+      this.get('ajax').ajax(`/v3/repo/${repo.get('id')}/unstar`, 'POST')
+        .then(() => {
+          this.get('flashes')
+            .success(`You successfully unstarred ${repo.get('slug')}`);
+        });
     }
```

Comment:
This is an unfortunate limitation with stock Ember Data, which is that “non-CRUD” actions aren’t really supported. Ember Data would like `starred` to be updated via the usual `PUT` (or `PATCH`?) to the endpoint (via `set`-ing the attribute and `save`ing), but the API wants a `POST` to `star` and `unstar`, which doesn’t match Ember Data conventions. (Hopefully I’m not just restating things you already know!)

I know of [`ember-data-actions`](https://github.com/davewasmer/ember-data-actions) for adding support for non-CRUD actions, or this [blog post showing a similar manual implementation](https://karolgalanciak.com/blog/2016/02/07/implementing-non-restful-actions-with-ember-data/). These may be worth looking more into to clean this up a little, but I’ve never used them myself…???

As a quick hack, you could use `repo.set('starred', true)` and `false` when starring and unstarring, which would cause the client to reflect the changes on the server. This would probably be best done within the `then` to guarantee that the change was successful. It makes me feel a bit 😳 but it really amounts to the same thing so I can push the 😳 away haha

Yet another option would be to call `repo.reload()` within the `then`, which would trigger a refresh from the API! This seems wasteful, having an entire API call refresh the model, but it’s more “clean” because it’s not manually mirroring the server- and client-side changes.

I like the first option the most because it would give us a nice `repo.star()` and `repo.unstar()` interface, which is the most Ember Data-y. It’s concise enough that I’d feel fine just having it within the component instead of the action needing to bubble up to the controller. However, this solution is also the most involved!

Well hopefully the above isn’t Too Much. Let me know if you want to discuss further!